sdk/couchbase-core/src/queryx/query.rs

```diff
@@ -138,4 +147,538 @@ impl<C: Client> Query<C> {
+
+        let mut qs = String::new();
+        if let Some(index_name) = &opts.index_name {
+            let encoded_name = encode_identifier(index_name);
+
+            if opts.scope_name.is_some() || opts.collection_name.is_some() {
+                qs.push_str(&format!("DROP INDEX {}", encoded_name));
+                qs.push_str(&format!(" ON {}", keyspace));
+            } else {
+                qs.push_str(&format!("DROP INDEX {}.{}", keyspace, encoded_name));
+            }
+        } else {
+            qs.push_str(&format!("DROP PRIMARY INDEX ON {}", keyspace));
+        }
+
+        let query_opts = QueryOptions::new()
+            .statement(qs)
+            .on_behalf_of(opts.on_behalf_of.clone());
+
+        let mut res = self.query(&query_opts).await;
+
+        match res {
+            Err(e) => match &*e.kind {
+                ErrorKind::Resource(resource_error) => match &*resource_error.cause.kind {
+                    ServerErrorKind::IndexNotFound => {
+                        if opts.ignore_if_not_exists.unwrap_or(false) {
+                            Ok(())
+                        } else {
+                            Err(e)
+                        }
+                    }
+                    _ => Err(e),
+                },
+                _ => Err(e),
+            },
+            Ok(_) => Ok(()),
+        }
+    }
+
+    pub async fn drop_index(&self, opts: &DropIndexOptions) -> error::Result<()> {
+        let encoded_name = encode_identifier(&opts.index_name);
+        let keyspace = build_keyspace(
+            opts.bucket_name.as_ref(),
+            &opts.scope_name.as_deref(),
+            &opts.collection_name.as_deref(),
+        );
+
+        let mut qs = String::new();
+        if opts.scope_name.is_some() || opts.collection_name.is_some() {
+            qs.push_str(&format!("DROP INDEX {}", encoded_name));
+            qs.push_str(&format!(" ON {}", keyspace));
+        } else {
+            qs.push_str(&format!("DROP INDEX {}.{}", keyspace, encoded_name));
+        }
+
+        let query_opts = QueryOptions::new()
+            .statement(qs)
+            .on_behalf_of(opts.on_behalf_of.clone());
+
+        let res = self.query(&query_opts).await;
+
+        match res {
+            Err(e) => match &*e.kind {
+                ErrorKind::Resource(resource_error) => match &*resource_error.cause.kind {
+                    ServerErrorKind::IndexNotFound => {
+                        if opts.ignore_if_not_exists.unwrap_or(false) {
+                            Ok(())
+                        } else {
+                            Err(e)
+                        }
+                    }
+                    _ => Err(e),
+                },
+                _ => Err(e),
+            },
+            Ok(_) => Ok(()),
+        }
+    }
+
+    pub async fn build_deferred_indexes(
+        &self,
+        opts: &BuildDeferredIndexesOptions,
+    ) -> error::Result<()> {
+        let opts = GetAllIndexesOptions {
```

Comment:
This makes me think that at this submodule level the fields on these options should be borrows. What do you think?